src/main/scala/mesosphere/marathon/health/MarathonHealthCheckManager.scala

```diff
@@ -28,14 +28,11 @@ class MarathonHealthCheckManager @Inject() (
     taskTracker: TaskTracker,
     appRepository: AppRepository) extends HealthCheckManager {
 
-  // composite key for partitioning the set of active health checks
-  protected[this] case class AppVersion(id: PathId, version: Timestamp)
-
   protected[this] case class ActiveHealthCheck(
     healthCheck: HealthCheck,
     actor: ActorRef)
 
-  protected[this] var appHealthChecks = Map[AppVersion, Set[ActiveHealthCheck]]()
+  protected[this] var appHealthChecks: Map[PathId, Map[Timestamp, Set[ActiveHealthCheck]]] = Map.empty
```

Comment:
Create a map with default Value, so you can get rid of all subsequent getOrlElse statements.